source/blog/2016-06-14-rspec-3-5-has-been-released.md

```diff
@@ -0,0 +1,368 @@
+* Some users have expressed surprised that the metadata is treated "special" here and
+  isn't simply applied to the shared example group like it is applied to a normal example group.
+* It makes it impossible to attach some metadata to a shared example group that will be
+  automatically applied to including example groups. For example, maybe you want to temporarily
+  add `:skip` or `:focus` metadata to all including groups. There was no way to do this.
+* There's no obvious way to make a shared example group get auto-included in every
+  example group (e.g. for a global `before` hook or a `let` you want to make available everywhere...).
+* It's inconsistent with how module inclusion works (e.g. `config.include DBSupport, :db`).
+
+In RSpec 3.5 we've rectified these problems with a couple of changes.
+
+#### New API: `config.include_context`
+
+You can now define shared context inclusions in your `RSpec.configure` block:
+
+~~~ ruby
+RSpec.configure do |config|
+  config.include_context "DB support", :db
+end
+~~~
+
+This aligns with the existing `config.include` API for module inclusions, provides
+a way to include shared contexts based on metadata that is less surprising, and
+makes it simple to include a shared context in all example groups (just don't
+pass a metadata argument).
+
+#### New config option: `config.shared_context_metadata_behavior`
+
+We've also added a config option that lets you determine how shared context
+metadata is treated:
+
+~~~ ruby
+RSpec.configure do |config|
+  config.shared_context_metadata_behavior = :trigger_inclusion
+  # or
+  config.shared_context_metadata_behavior = :apply_to_host_groups
+end
+~~~
+
+The former value (`:trigger_inclusion`) is the default and exists only for backwards
+compatibility. It treats metadata passed to `RSpec.shared_context` exactly how it was
+treated in RSpec 3.4 and before: it triggers inclusion in groups with matching metadata.
+We plan to remove support for it in RSpec 4.
+
+The latter value (`:apply_to_host_groups`) opts-in to the new behavior. Instead of
+triggering inclusion in groups with matching metadata, it applies the metadata to host
+groups.  For example, you could focus on all groups that use the DB by tagging your
+shared context:
+
+~~~ ruby
+RSpec.shared_context "DB support", :focus do
+  # ...
+end
+~~~
+
+### Expectations: Keyword argument support for the `respond_to` matcher.
+
+Keyword arguments have been a stable language feature for some time now but RSpec
+has lacked support for placing expectations upon keyword arguments in a variety of our matchers.
+
+In rspec-expectations 3.5 we add support for checking whether an object responds to a method
+using keyword arguments. You'll now be able to check a methods response signature for particular
+keywords as well as arguments.
+
+~~~ ruby
+  expect(my_object).to respond_to(:find).with_keywords(:limit, :offset) }
+  expect(my_object).to respond_to(:find).with(1).argument.and_keywords(:limit, :offset)
```

Comment:
One thing that's unclear to me in this example (and that perhaps warrants changes in rspec-expectations before we are locked into the semantics) is what it means to mix an argument count and keywords.  Is the `(1).argument` bit specifying the number of non-keyword arguments (e.g. positional arguments)?  Where it gets confusing is that at the call site keyword args are just passed as a single hash arg, which is then interpreted as keyword args or as a single positional arg depending on how the method is defined.  So it's not clear to me if keyword args should be counted and if so, if they should be counted as one arg (since the underlying mechanism uses a single hash) or as one arg per keyword.